cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+				remotes[i] = r.URLs[0]
+				i++
+			}
+			repoURL, err = app.Prompt.CaptureList("Which is the remote URL for this repo?", remotes)
+			if err != nil {
+				// TODO Would we really want to abort here?
+				return err
+			}
+			return nil
+		}
+		repoURL, err = app.Prompt.CaptureString("Please provide the repository URL")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// loadYAMLFile loads a YAML file from disk into a concrete types.Definition object
+// using generics. It's role really is solely to verify that the YAML content is valid.
+func loadYAMLFile[T types.Definition](path string, defType T) error {
+	fileBytes, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	return yaml.Unmarshal(fileBytes, &defType)
+}
+
+func getSubnetInfo(sc *models.Sidecar) (*types.Subnet, error) {
+	homepage, err := app.Prompt.CaptureEmpty("What is the homepage of the Subnet project?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := app.Prompt.CaptureEmpty("Please provide a free-text description of the Subnet", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	maintrs, canceled, err := app.Prompt.CaptureListDecision(
+		app.Prompt,
+		"Who are the maintainers of the Subnet?",
+		app.Prompt.CaptureEmail,
+		"Provide a maintainer",
+		"Maintainer",
+		"",
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	if canceled {
+		ux.Logger.PrintToUser("Publishing aborted")
+		return nil, errors.New("canceled by user")
+	}
+
+	strMaintrs := make([]string, len(maintrs))
+	for i, m := range maintrs {
+		strMaintrs[i] = m.(string)
+	}
+
+	// TODO: In this version, we are publishing 1 subnet and exactly 1 VM.
+	// 1. Will the VM list has to be editale in the future?
+	// 2. If yes, need to think about the whole process
+	// 3. Can VMs later be added independently?
+	/*
+		vms, canceled, err := app.Prompt.CaptureListDecision(
+			app.Prompt,
+			"Provide a list of VMs this Subnet is running",
+			app.Prompt.CaptureEmpty,
+			"Provide a VM",
+			"VM",
+			"VMs are instances of blockchains a given Subnet is running.",
+			nil,
+		)
+		if err != nil {
+			return nil, err
+		}
+		if canceled {
+			ux.Logger.PrintToUser("Publishing aborted")
+			return nil, errors.New("canceled by user")
+		}
+
+		strVMs := make([]string, len(vms))
+		for i, v := range vms {
+			strVMs[i] = v.(string)
+		}
+	*/
+
+	subnet := &types.Subnet{
+		ID:          sc.Networks[models.Fuji.String()].SubnetID.String(),
+		Alias:       sc.Name,
+		Homepage:    homepage.(string),
+		Description: desc.(string),
+		Maintainers: strMaintrs,
+		// VMs:         strVMs,
+		VMs: nil,
+	}
+
+	return subnet, nil
+}
+
+func getVMInfo(sc *models.Sidecar) (*types.VM, error) {
+	vmID, err := app.Prompt.CaptureEmpty("What is the ID of this VM?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := app.Prompt.CaptureEmpty("Provide a description for this VM please", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	maintrs, canceled, err := app.Prompt.CaptureListDecision(
```

Comment:
I'm not 100% sure of this, but do we maybe only want to capture one set of maintainers and use them in both places?

Reply:
Alternatively, should we add ava labs if using spaces or subnet-evm?

Reply:
I thought of these questions too and couldn't get to a definitive answer either. 

Like - I don't think we can just provide the same maintainers. 
However, I don't like to have to ask nearly the same question twice either.

Maybe we should do as above - if CustomVM, ask twice, otherwise just use spaces and subnet-evm. Probably what you had in mind too.

Reply:
that sounds great!